### PR TITLE
feat: add yarn spinner

### PR DESCRIPTION
**Pull request by kaisoapbox**
## Description
What changes does this PR include? Is there anything that affects other features that people should be aware of?
- added yarn spinner integration
- added beginnings of story dialogue stuff
- added textures for text box, avatar switching with the character, etc
- imported a whole bunch of fonts and things
- also made sure that the line view scrolling whatever stuff interacts fine with `timeScale=0` for pausing

## Before and After
Screenshots/videos of the changes:

https://github.com/user-attachments/assets/a15d4e6d-6e41-4199-8a12-4f348e293771

## Checklist
- [x] This code builds and runs
- [x] All public classes and methods are documented
- [x] Hard-to-understand areas of the code are documented
- [x] Self-review done

### CODE REVIEW

**Comment by kaisoapbox** [comment]
image

just realized i didnt test how it looks with multi-line text, how do we feel about this? i think it's ok but it might be slightly better if the avatar were centered?

**Comment by jentaruno** [comment]
> image
> just realized i didnt test how it looks with multi-line text, how do we feel about this? i think it's ok but it might be slightly better if the avatar were centered?

Agree, I think centering the avatar wld fix it

**Comment by kaisoapbox** [comment]
image

fixy

**Review by jentaruno — CHANGES_REQUESTED** [review]
I like what u did with Effects --> UnscaledEffects, good stuff.
Only comment is on Checkpoint.cs

File: Assets/Scripts/Checkpoint.cs [hunk 1]
```diff
@@ -1,25 +1,73 @@
...
+    [Header("References")] [SerializeField]
+    private Animator anim;
+
+    [SerializeField] private SpriteRenderer sprite;
+    [SerializeField] private string conversationStartNode;
+    [SerializeField] private bool beginsInteractable;
+
+    // internal properties not exposed to editor
+    private DialogueRunner dialogueRunner;
+    private bool interactable;
```
[the remark this answers is not in the document]
Comment (kaisoapbox) [reply Assets/Scripts/Checkpoint.cs]: oh right i wrote this code before setting up rider, woops
fixy'd

File: Assets/Scripts/Yarn/LineView.cs [hunk 2]
```diff
@@ -297,6 +297,12 @@ private void Reset()
             canvasGroup = GetComponentInParent<CanvasGroup>();
         }
 
+        private void Update()
+        {
+            // if hitting spacebar during update, trigger dialogue advancement
+            if (Input.GetButtonDown("Jump")) UserRequestedViewAdvancement();
+        }
+
```
Comment (kaisoapbox) [inline Assets/Scripts/Yarn/LineView.cs:305]: @jentaruno i just added this too, it allows spacebar to basically trigger a continue button press

Comment (jentaruno) [reply Assets/Scripts/Yarn/LineView.cs:305]: BOOTS

**Review by jentaruno — CHANGES_REQUESTED** [review]
Oh I didn't think you'd remove interactable, i thought you'd actually make it work lol if us devs want to disable the dialog!

Looking at it again though we could just set Conversation Start Node to "" and not add an extra field to disable the dialog. If this is the case, then we should add a comment to conversationStartNode describing what it does and how setting it to an empty string would make this checkpoint skip a dialog.

Also beginsInteractable is still unused

**Comment by kaisoapbox** [comment]
ngl i just forget what interactable was meant to do so i just dunked it lmao

**Review by jentaruno — APPROVED** [review]
Yassss

**Pull request merged**